pkg/operation/botanist/controlplane/kubescheduler/kube_scheduler.go

```diff
@@ -0,0 +1,473 @@
+					},
+					{
+						Name: k.secrets.Server.Name,
+						VolumeSource: corev1.VolumeSource{
+							Secret: &corev1.SecretVolumeSource{
+								SecretName: k.secrets.Server.Name,
+							},
+						},
+					},
+					{
+						Name: configMap.Name,
+						VolumeSource: corev1.VolumeSource{
+							ConfigMap: &corev1.ConfigMapVolumeSource{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: configMap.Name,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (k *kubeScheduler) SetSecrets(secrets Secrets)          { k.secrets = secrets }
+func (k *kubeScheduler) Destroy(_ context.Context) error     { return nil }
+func (k *kubeScheduler) Wait(_ context.Context) error        { return nil }
+func (k *kubeScheduler) WaitCleanup(_ context.Context) error { return nil }
+
+func (k *kubeScheduler) emptyConfigMap() *corev1.ConfigMap {
+	return &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "kube-scheduler-config", Namespace: k.namespace}}
+}
+
+func (k *kubeScheduler) emptyVPA() *autoscalingv1beta2.VerticalPodAutoscaler {
+	return &autoscalingv1beta2.VerticalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: "kube-scheduler-vpa", Namespace: k.namespace}}
+}
+
+func (k *kubeScheduler) emptyService() *corev1.Service {
+	return &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "kube-scheduler", Namespace: k.namespace}}
+}
+
+func (k *kubeScheduler) emptyDeployment() *appsv1.Deployment {
+	return &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.DeploymentNameKubeScheduler, Namespace: k.namespace}}
+}
+
+func (k *kubeScheduler) computeServerPort() int32 {
+	if k.versionConstraints.k8sGreaterEqual113 {
+		return 10259
+	}
+	return 10251
+}
+
+func (k *kubeScheduler) computeServerURIScheme() corev1.URIScheme {
+	if k.versionConstraints.k8sGreaterEqual113 {
+		return corev1.URISchemeHTTPS
+	}
+	return corev1.URISchemeHTTP
+}
+
+func (k *kubeScheduler) computeEnvironmentVariables() []corev1.EnvVar {
+	if k.config != nil && k.config.KubeMaxPDVols != nil {
+		return []corev1.EnvVar{{
+			Name:  "KUBE_MAX_PD_VOLS",
+			Value: *k.config.KubeMaxPDVols,
+		}}
+	}
+	return nil
+}
+
+func (k *kubeScheduler) computeComponentConfig() (string, string, error) {
+	var apiVersion string
+	if k.versionConstraints.k8sGreaterEqual118 {
+		apiVersion = "kubescheduler.config.k8s.io/v1alpha2"
+	} else if k.versionConstraints.k8sGreaterEqual112 {
+		apiVersion = kubeschedulerv1alpha1.SchemeGroupVersion.String()
+	} else if k.versionConstraints.k8sGreaterEqual110 {
+		apiVersion = "componentconfig/v1alpha1"
+	}
+
+	componentConfig := &kubeschedulerv1alpha1.KubeSchedulerConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: apiVersion,
+			Kind:       "KubeSchedulerConfiguration",
+		},
+		ClientConnection: componentbaseconfigv1alpha1.ClientConnectionConfiguration{
+			Kubeconfig: volumeMountPathKubeconfig + "/" + secrets.DataKeyKubeconfig,
```

Comment:
Just a suggestion :)
Can be applied to all the other paths (in the command line flags) too, if you want.
```suggestion
			Kubeconfig: filepath.Join(volumeMountPathKubeconfig, secrets.DataKeyKubeconfig),
```

Reply:
Wouldn't this break if you develop/build Gardener on e.g. Windows?

Reply:
Oh well, I missed that. Just forget my suggestion, then 😄